src/modules/land_detector/LandDetector.cpp

```diff
@@ -158,16 +157,22 @@ void LandDetector::_check_params(const bool force)
 {
 	parameter_update_s paramUpdate;
 
-	if (_parameterSub.update(&paramUpdate) || force) {
+	if (_parameter_sub.update(&paramUpdate) || force) {
 		_update_params();
 		uint32_t flight_time;
-		param_get(_p_total_flight_time_high, (int32_t *)&flight_time);
+		param_get(_paramHandle.total_flight_time_high, (int32_t *)&flight_time);
 		_total_flight_time = ((uint64_t)flight_time) << 32;
-		param_get(_p_total_flight_time_low, (int32_t *)&flight_time);
+		param_get(_paramHandle.total_flight_time_low, (int32_t *)&flight_time);
```

Comment:
Why is this duplicated with `_update_params()`?

Reply:
Hi @julianoes , thanks for pointing that out.  I think I don't actually understand what the `_update_params()` call at line 161 is doing right now in the current master branch, (not changed in this PR), as it appears to be doing nothing.

I've pushed up a commit to clean that up.  Would you mind looking closely at that with me in the current master branch as well as this PR?  Thanks!

Reply:
The last commit I pushed up has an issue that shows up in flight testing.  I am reverting it back momentarily.

Reply:
The commit changing to `ScheduleNow()` caused the issue I saw.  I've reverted back to my first commit, that uses `ScheduleOnInterval()` and flight tested.

Reply:
Ok, I've got things resolved.  Changing to `ScheduleNow()` required rescheduling in the `Run()` method.  I don't know why `ScheduleOnInterval()` doesn't play well with JMavSim, something we might need to look into.

Most recent commit fixes issue 12190 and is flight tested here: https://review.px4.io/plot_app?log=70bc2060-629d-4b26-8b0f-b5d0bc00e9fb